lib/jobs/ucs-discovery.js

```diff
@@ -140,44 +140,57 @@ function UcsDiscoveryJobFactory(
                         return res.body;
                     })
                     .map(function(data) {
-                        assert.array(data.macs);
-                        assert.string(data.name);
-                        assert.string(data.path);
-
-                        var config = Object.assign({}, self.settings);
-                        obm = {
-                            config: config,
-                            service: 'ucs-obm-service'
-                        };
-                        var node = {
-                            type: 'compute',
-                            name: data.path,
-                            identifiers: data.macs,
-                            obm:[obm],
-                            relations : []
-                        };
-
-                        // Update existing node  or create a new one
-                        return waterline.nodes.needOne({identifiers:data.macs})
-                            .then(function(curNode) {
-                                return waterline.nodes.updateOne(
-                                    { id: curNode.id },
-                                    node
-                                )
-                                .then(function(data){
-                                    return data;
+                        self.createNode(data, 'compute');
+                    });
+            }else{
+                logger.warning('No rackmount servers found Found');
+                return;
+            }
+        });
+    };
+
+    /**
+     * @function createServers
+     * @description discovers all the chassis servers in the Cisco UCS
+     */
+    UcsDiscoveryJob.prototype.createServers = function (root) {
```

Comment:
Too many nesting levels for this function! it's hard to read, please split it into smaller functions to reduce the nesting levels.